File: java/application/src/main/java/dev/ufadevconf/application/CreatePickupPointCommand.java
package dev.ufadevconf.application;

import java.util.Objects;

public record CreatePickupPointCommand(String code, String address, long minWeightGrams, long maxWeightGrams) {
    public CreatePickupPointCommand {
        Objects.requireNonNull(code, "code");
        Objects.requireNonNull(address, "address");
        if (minWeightGrams < 0) {
            throw new IllegalArgumentException("Minimum weight cannot be negative");
        }
        if (maxWeightGrams < 0) {
            throw new IllegalArgumentException("Maximum weight cannot be negative");
        }
    }
}

File: kotlin/application/src/main/kotlin/dev/ufadevconf/application/CreatePickupPointCommand.kt
package dev.ufadevconf.application

data class CreatePickupPointCommand(
    val code: String,
    val address: String,
    val minWeightGrams: Long,
    val maxWeightGrams: Long
) {
    init {
        require(minWeightGrams >= 0) { "Minimum weight cannot be negative" }
        require(maxWeightGrams >= 0) { "Maximum weight cannot be negative" }
    }
}
